WalletWasabi.Fluent/Models/Wallets/WalletTransactionsModel.cs

```diff
@@ -45,6 +46,22 @@ public bool AreEnoughToCreateTransaction(TransactionInfo transactionInfo, IEnume
 		return TransactionHelpers.TryBuildTransactionWithoutPrevTx(_wallet.KeyManager, transactionInfo, _wallet.Coins, coins, _wallet.Kitchen.SaltSoup(), out _);
 	}
 
+	public async Task<TransactionSummary?> GetById(string transactionId)
+	{
+		var txRecordList = await Task.Run(() => TransactionHistoryBuilder.BuildHistorySummary(_wallet));
+
+		var transaction = txRecordList.FirstOrDefault(x => x.GetHash().ToString() == transactionId);
+		return transaction;
+	}
```

Comment:
Since `WalletTransactionsModel` has all the transactions, and keeps that list always up-to-date, this method could simply return the `TransactionSummary` object from the list, instead of building the whole history and finding the item.

Since it is as of now equivalent to what we have on master, this is not a blocker, but fix it in a separate PR.
@ichthus1604 @SuperJMN